builder/bundles.go

```diff
@@ -840,6 +841,79 @@ func buildFullChroot(b *Builder, set *bundleSet, packagerCmd []string, buildVers
 		}
 	}
 
+	// Resolve bundle content chroots against the full chroot. New files are copied
+	// to the full chroot and the bundle file lists are updated to contain the files
+	// within the chroot.
+	return addBundleContentChroots(set, fullDir)
```

Comment:
In case the user provides the os-core bundle also as content that contains the os-release file, should we call `buildOsCore()` and `genUpdateBundleSpecialFiles()` after `addBundleContentChroots()`. i.e take them outside the for loop and have something like this:

```
	if err := addBundleContentChroots(set, fullDir); err != nil {
		return err
	}

	// special handling for os-core
	fmt.Println("... building special os-core content")
	if err := buildOsCore(b, packagerCmd, fullDir, version); err != nil {
		return err
	}

	// special handling for update bundle
	fmt.Printf("... Adding swupd default values to %s bundle\n", b.Config.Swupd.Bundle)
	if err := genUpdateBundleSpecialFiles(fullDir, b); err != nil {
		return err
	}

	return err
```

Reply:
Are you saying that users should be allowed to override these special case files and prevent Mixer from auto-generating them?

Reply:
What I meant is, in case the user provides the special case files, then it gets updated by Mixer after the content chroot is created.

Reply:
As discussed, lets give the content more priority. So if the special case files exist as content, then do not overwrite them. The fields in os-release should still be updated.

Reply:
> In case the user provides the os-core bundle also as content that contains the os-release file, should we call `buildOsCore()` and `genUpdateBundleSpecialFiles()` after `addBundleContentChroots()`. i.e take them outside the for loop and have something like this:

Please ignore this comment and let Mixer fail if user provides conflicting special case files as contents. The reason being:
Swupd_Root.pem is a special case file that is either created by the Mixer or specified by the user using the Builder.CERT config. This certificate file is used by Mixer for signing. So if we let the user overwrite it as a content file, it will lead to an inconsistent mix as the certificate used and certificate in the mix will be different. One way to handle this is to treat this special case file as even more special. But lets not make it more complicated and hard to maintain. @jwakre Sorry for the additional work.

Reply:
> As discussed, lets give the content more priority. So if the special case files exist as content, then do not overwrite them. The fields in os-release should still be updated.

Please ignore this comment and let Mixer fail if user provides conflicting special case files as contents. The reason being:
Swupd_Root.pem is a special case file that is either created by the Mixer or specified by the user using the Builder.CERT config. This certificate file is used by Mixer for signing. So if we let the user overwrite it as a content file, it will lead to an inconsistent mix as the certificate used and certificate in the mix will be different. One way to handle this is to treat this special case file as even more special. But lets not make it more complicated and hard to maintain. @jwakre Sorry for the additional work.